parsers/saarland.py

```diff
@@ -0,0 +1,223 @@
+
+def parse_url(url, today=False):
+    global base_data
+
+    load_base_data()
+
+    canteen = LazyBuilder()
+    with urlopen(url) as response:
+        data = json.loads(response.read().decode())
+
+    for day in data['days']:
+        date = datetime.datetime.strptime(day['date'], UTC_DATE_STRING).date()
+
+        if today and (datetime.date.today() != date):
+            continue
+
+        for counter in day['counters']:
+            counter_name = counter['displayName']
+            counter_description = counter['description']
+            counter_hours = counter.get('openingHours')
+
+            for meal in counter['meals']:
+                if 'knownMealId' in meal:
+                    # This is meant to allow recognizing recurring meals,
+                    # for features like marking meals as favorites.
+                    # Up to now, not really used in the mensaar.de API.
+                    print('knownMealId: %s' % meal['knownMealId'])
+
+                meal_name = meal['name']
+                if 'category' in meal:
+                    meal_name = '%s: %s' % (meal['category'], meal_name)
+
+                meal_notes = (
+                    build_location(counter_description) +
+                    build_hours(counter_hours) +
+                    build_notes(meal['notices'], meal['components']))
+
+                meal_prices = {}
+                if 'prices' in meal:
+                    prices = meal['prices']
+                    for role in prices:
+                        if role in base_data['roles']:
+                            meal_prices[base_data['roles'][role]] = prices[role]
+                        else:
+                            print('Ignoring price for unknown role %s.' % role)
+
+                if 'pricingNotice' in meal:
+                    meal_notes.append(meal['pricingNotice'])
+
+                canteen.addMeal(date, counter_name,
+                                meal_name, meal_notes, meal_prices)
+
+    return canteen.toXMLFeed()
+
+
+def load_base_data():
+    with urlopen(URL_BASE + URL_BASE_DATA) as response:
+        data = json.loads(response.read().decode())
+
+    base_data['roles'] = {}
+
+    base_data['notices'] = data['notices']
+    for loc in data['locations']:
+        if loc not in LOCATIONS:
+            # Found an unknown location
+            # Please consider updating the parser!
+            raise RuntimeError(
+                'Unknown location: %s (displayName: %s)' %
+                (loc, data['locations'][loc]))
+    for role in data['priceTiers']:
+        if role not in ROLES:
+            # Found an unknown price tier
+            # All prices for this price tier will be ignored.
+            # Please consider updating the parser!
+            raise RuntimeError(
+                'Unknown price tier: %s (displayName: %s)' %
+                (role, data['priceTiers'][role]))
+        else:
+            base_data['roles'][role] = ROLES[role]
+
+
+base_data = {}
```

Comment:
Is there is specific reason for storing this data in a global variable? If not I would prefer just returning the data from the function and passing it to the other functions as needed.

Reply:
No, there is no reason. I've changed it.